attachments do not get created from 2nd and subsequent tests

I am trying to use the attachment feature to attach a few log files. When I start my test run, I see attachment getting created in the first test. However, from second test onwards I see an error:
**_log() got an unexpected keyword argument 'attachment'** 

Following is the code snippet I am trying:
```
def _set_portal_logging(service):
    """
    Report portal logger
    """
    logging.setLoggerClass(RPLogger)
    logger = logging.getLogger()
    logger.setLevel(logging.INFO)
    rp_handler = RPLogHandler(service)
    rp_handler.setLevel(logging.INFO)
    # logger.addHandler(rp_handler)
    return logger

@pytest.fixture(scope='function')
def ips_oprofile_reports(request, module_setup, commandline_args):
    """
    Fixture for downloading oprofile reports from
    oprofile server and attaching it to report portal
    logs
    """
    _set_portal_logging(request.node.config.py_test_service)
    args = commandline_args
    ips_api = ips_module_setup
    yield
    if ips_api.result is False:
        logging.info("Test failed in ixload configuration")
    else:
       for file in report_list:
                try:
                    content = profile.download_report_file(id, file)
                    if len(content) < 1:
                        continue
                    logging.info("attaching file {}_{}".format(file, id),
                                attachment={"name": '{}_{}'.format(file, id),
                                            "data": content,
                                            "mime": "application/octet-stream",}
                                ,)
                except Exception as e:
                    logging.info(e)
                    continue

```

As example, 

conftest.py file content:
```@pytest.fixture(scope="session")
def rp_logger(request):
    import logging
    logger = logging.getLogger(__name__)
    logger.setLevel(logging.DEBUG)
    # Create handler for Report Portal if the service has been
    # configured and started.
    if hasattr(request.node.config, 'py_test_service'):
        # Import Report Portal logger and handler to the test module.
        from pytest_reportportal import RPLogger, RPLogHandler
        logging.setLoggerClass(RPLogger)
        rp_handler = RPLogHandler(request.node.config.py_test_service)
    else:
        import sys
        rp_handler = logging.StreamHandler(sys.stdout)
    # Set INFO level for Report Portal handler.
    rp_handler.setLevel(logging.INFO)
    return logger

def pytest_generate_tests(metafunc):
    if metafunc.function.__name__ == "test_rp":
        metafunc.parametrize("tc", [True, True])
```

rp_test.py file content:
```
def test_rp(request, tc, rp_logger):
    html = """<html lang="utf-8">
     <body><p>Paragraph</p></body></html>"""
    rp_logger.info("Html report",
                   attachment={"name": "Detailed report",
                               "data": html,
                               "mime": "text/html"})
    assert tc
```
Result:
`====================== 1 failed, 1 passed in 0.40 seconds ======================`
```
        if self.isEnabledFor(INFO):
>           self._log(INFO, msg, args, **kwargs)
E           TypeError: _log() got an unexpected keyword argument 'attachment'
```

--- a/README.rst
+++ b/README.rst
@@ -95,6 +95,14 @@ in conftest.py:
 
 .. code-block:: python
 
+    import logging
+    import sys
+
+    import pytest
+
+    from pytest_reportportal import RPLogger, RPLogHandler
+
+
     @pytest.fixture(scope="session")
     def rp_logger(request):
         import logging
@@ -104,11 +112,13 @@ in conftest.py:
         # configured and started.
         if hasattr(request.node.config, 'py_test_service'):
             # Import Report Portal logger and handler to the test module.
-            from pytest_reportportal import RPLogger, RPLogHandler
             logging.setLoggerClass(RPLogger)
             rp_handler = RPLogHandler(request.node.config.py_test_service)
+            # Add additional handlers if it is necessary
+            console_handler = logging.StreamHandler(sys.stdout)
+            console_handler.setLevel(logging.INFO)
+            logger.addHandler(console_handler)
         else:
-            import sys
             rp_handler = logging.StreamHandler(sys.stdout)
         # Set INFO level for Report Portal handler.
         rp_handler.setLevel(logging.INFO)

--- a/pytest_reportportal/rp_logging.py
+++ b/pytest_reportportal/rp_logging.py
@@ -142,14 +142,14 @@ def patching_logger_class():
                 return record
             return makeRecord
 
-        if not logger_class == RPLogger and not hasattr(logger_class, "_patched"):
+        if not issubclass(logger_class, RPLogger):
             logger_class._log = wrap_log(logger_class._log)
             logger_class.makeRecord = wrap_makeRecord(logger_class.makeRecord)
-            logger_class._patched = True
+            logging.setLoggerClass(RPLogger)
         yield
 
     finally:
-        if hasattr(logger_class, '_patched'):
+        if not issubclass(logger_class, RPLogger):
             logger_class._log = original_log
             logger_class.makeRecord = original_makeRecord
-            del logger_class._patched
+            logging.setLoggerClass(logger_class)

